packages/app/src/app/components/CreateSandbox/Import/useGitHubAccountRepositories.ts

```diff
@@ -0,0 +1,87 @@
+import { useQuery } from '@apollo/react-hooks';
+import {
+  GetGitHubAccountReposQuery,
+  GetGitHubAccountReposQueryVariables,
+  GetGitHubOrganizationReposQuery,
+  GetGitHubOrganizationReposQueryVariables,
+} from 'app/graphql/types';
+import {
+  GET_GITHUB_ACCOUNT_REPOS,
+  GET_GITHUB_ORGANIZATION_REPOS,
+} from '../queries';
+
+// GitHub makes a distinction between personal and organization accounts
+// both are accounts, so I'm calling them that.
+
+type UseGitHubAccountRepositoriesOptions = {
+  name?: string;
+  accountType?: 'personal' | 'organization';
+};
+
+export const useGitHubAccountRepositories = ({
+  name,
+  accountType,
+}: UseGitHubAccountRepositoriesOptions) => {
+  const skipLoadingPersonal = accountType === 'organization' || !name;
+  const skipLoadingOrganization = accountType === 'personal' || !name;
+
+  // Query the personal repositories unless the selected github account
+  // is an organization
+  const account = useQuery<
+    GetGitHubAccountReposQuery,
+    GetGitHubAccountReposQueryVariables
+  >(GET_GITHUB_ACCOUNT_REPOS, {
+    skip: skipLoadingPersonal,
+    variables: {
+      perPage: 10, // TODO determine how much repos
+      page: 1,
+    },
+    // Apollo (this version) has weird caching issues where fetching anything from "me" with apollo overrides
```

Comment:
Can you specify the version number?

Reply:
Done in [9a30997](https://github.com/codesandbox/codesandbox-client/pull/7436/commits/9a309978d424f364c5e3d5e45b0eda4388f2ee2a).